### DISCUSSION TITLE
Help deploying Python Flask app

### DISCUSSION THREAD

**Discussion by WhiteHeadbanger** [vercel/community#4555] [Help]
### Summary

I'm trying to deploy my app to Vercel, to no success.

It runs perfectly on local.

I've tried a lot of things, please see below. 

### Steps to Reproduce

Proyect structure:
![image](https://github.com/vercel/community/assets/12226415/7b78954e-cada-4f8e-8a1d-00bd94003b4a)

**Attempt n° 1**
Build and deployment settings:
![image](https://github.com/vercel/community/assets/12226415/e979ceeb-2ae6-4e12-80ac-6c1c450886d5)

Result:
`Error: No more than 12 Serverless Functions can be added to a Deployment on the Hobby plan. Create a team (Pro plan) to deploy more. Learn More: https://vercel.link/function-count-limit`

**Attempt n° 2**

Using vercel.json:
```
{
    "version": 2,
    "builds": [
        {
            "src": "./api/app.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/api/(.*)",
            "dest": "/api"
        }
    ]
}
```

Result:
`WARN! Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings`

![image](https://github.com/vercel/community/assets/12226415/6398e7ba-2b6c-41a3-8a14-6196ebf203f3)

**Attempt n° 3**
```
{
    "functions": {
        "api/app.py": {
            "memory": 1024,
            "maxDuration": 10
        }
    },
    "installCommand": "pip install -r requirements.txt"
}
```

Result:
`Error: No more than 12 Serverless Functions can be added to a Deployment on the Hobby plan. Create a team (Pro plan) to deploy more. Learn More: https://vercel.link/function-count-limit`

**Attempt n° 4**
![image](https://github.com/vercel/community/assets/12226415/af0ed4ae-09c0-4735-8cb8-2ed8044869af)

But it never finishes deployment, so I'm guessing this isn't the right direction, but at this point I'm just trying every combination that I can think of in the vercel.json file.

I don't know what else to do, please help 
____________________________________
**Update**

**Attempt n° 5**

```
{
    "builds": [
        {
            "src": "api/app.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "api/app.py"
        }
    ]
}
```

Result:
![image](https://github.com/vercel/community/assets/12226415/0c486d8f-588e-412e-af3c-a4d5e29d6ca3)

**Update n° 2**
Same vercel.json as before but I fixed some imports that Vercel wasn't recognizing. Now I have a Not found error:
![image](https://github.com/vercel/community/assets/12226415/c406bcb7-2860-489f-9943-215d67482d32)

I'm guessing Vercel can't find my routes? I have two blueprints, one for the app itself, the other for auth stuff. It doesn't matter which endpoint I request, the result is the same

**Update n° 3**
I finally did it! 
Steps to succeed:

- DON'T use `if __name__ == '__main__'`
- DON'T use `app.run()` or other libraries
- You CAN use `create_app()`, but I didn't. I just put everything that I needed on the app.py file:

```
from flask import Flask
from flask_login import LoginManager
from flask_migrate import Migrate
from flask_bcrypt import Bcrypt
from ._config import config
from .database.db import db

app = Flask(__name__)
_bcrypt = Bcrypt(app)

# Routes
from .routes import blog, auth

# Config
app.config.from_object(config['production'])

#database
db.init_app(app)

# Flask-login
login_manager = LoginManager()
login_manager.login_view = 'auth.login'
login_manager.init_app(app)

from .models.UserModel import UserModel

@login_manager.user_loader
def load_user(user_id):
    return UserModel.query.get(user_id)

# Migrate
migrate = Migrate(app, db)

with app.app_context():
    db.create_all()

# Blueprints
app.register_blueprint(blog.main, url_prefix='/blog')
app.register_blueprint(auth.auth, url_prefix='/auth')
```

and that's it

**Comment by LKing12** [comment]
Any idea on how to increase the maxDuration with this?

**Comment by aniket-00** [comment]
{
  "rewrites": [
    { "source": "/(.*)", "destination": "/api/app" }
  ]
}

try this in your vercel.json file
